packages/sanity/src/_internal/cli/actions/backup/resolveApiClient.ts

```diff
@@ -0,0 +1,44 @@
+import {CliCommandContext} from '@sanity/cli'
+import {SanityClient} from '@sanity/client'
+import {chooseDatasetPrompt} from '../dataset/chooseDatasetPrompt'
+
+type ResolvedApiClient = {
+  projectId: string
+  datasetName: string
+  token?: string
+  client: SanityClient
+}
+
+async function resolveApiClient(
+  context: CliCommandContext,
+  datasetName: string,
+  apiVersion: string,
+): Promise<ResolvedApiClient> {
+  const {apiClient} = context
+
+  let client = apiClient()
+  const {projectId, token} = client.config()
+
+  if (!projectId) {
+    throw new Error('Project ID not defined')
+  }
+
+  // Do not use dataset configured in Sanity config since dataset specified in command should override.
```

Comment:
Should we let the dataset in the config be the default?

Reply:
`sanity dataset export` does not use config dataset, so I followed the same pattern. A missing dataset name in config results in a dummy name which is a weird experience for backup/export.

Reply:
That makes sense 👍